x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/field_select.tsx

```diff
@@ -97,25 +100,31 @@ export function FieldSelect({
                   ? selectedColumnOperationType
                   : undefined,
             },
+            exists: fieldMap[field].exists || false,
             compatible: isCompatibleWithCurrentOperation(field),
           }))
-          .sort(({ compatible: a }, { compatible: b }) => {
-            if (a && !b) {
+          .filter(field => showEmptyFields || field.exists)
```

Comment:
It feels a bit strange to have the setting in the data panel take effect here. If a user is in the popover editor and wants to select a fields they know exists but the existence filter is on, it won't show up and the user has to close the editor, go over to the data panel and deactivate the filter there. It gets more confusing because the type filters (string, number, date) in the data panel don't take effect here. Maybe we should just put them down to the end of the list?

Reply:
I changed this based on early feedback that it was confusing to have two ways of doing this. I feel pretty strongly that we have a good default- hiding "empty" fields is going to help in a lot of common cases. I don't intend to change this right away, but if we get more feedback about this it's easy to add more control